incubator/hnc/internal/validators/anchor.go

```diff
@@ -90,7 +90,18 @@ func (v *Anchor) handle(req *anchorRequest) admission.Response {
 	}
 
 	if req.op == v1beta1.Delete {
-		if cns.Exists() && !cns.AllowsCascadingDelete() {
+		// If the subnamespace has full namespaces in the descendants, deny the request and ask users
+		// to remove the full descendant namespaces first.
+		fd := cns.FullDescendantNames()
+		if fd != nil {
+			msg := fmt.Sprintf("The subnamespace %s has full namespaces in the descendants. Please remove them first. "+
+				"Full descendant namespaces: %q.", cnm, fd)
+			return deny(metav1.StatusReasonForbidden, msg)
+		}
+
```

Comment:
This seems to belong in a different CL? This is about preventing orphans, not making it easier to delete leaves, right?

Reply:
Right, I removed this part and will do it in a separate PR.